extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/tuple/ArrayOfDoublesSketchAggregatorFactory.java

```diff
@@ -0,0 +1,319 @@
+        metricColumns.size(),
+        this.numberOfValues
+      );
+    }
+  }
+
+  @Override
+  public Aggregator factorize(final ColumnSelectorFactory metricFactory)
+  {
+    if (metricColumns == null) {
+      final BaseObjectColumnValueSelector<ArrayOfDoublesSketch> selector = metricFactory
+          .makeColumnValueSelector(fieldName);
+      if (selector instanceof NilColumnValueSelector) {
+        return new ArrayOfDoublesSketchNoOpAggregator(numberOfValues);
+      }
+      return new ArrayOfDoublesSketchMergeAggregator(selector, nominalEntries, numberOfValues);
+    }
+    final DimensionSelector keySelector = metricFactory
+        .makeDimensionSelector(new DefaultDimensionSpec(fieldName, fieldName));
+    if (keySelector == null || DimensionSelectorUtils.isNilSelector(keySelector)) {
+      return new ArrayOfDoublesSketchNoOpAggregator(numberOfValues);
+    }
+    final List<BaseDoubleColumnValueSelector> valueSelectors = new ArrayList<>();
+    for (final String column : metricColumns) {
+      final BaseDoubleColumnValueSelector valueSelector = metricFactory.makeColumnValueSelector(column);
+      valueSelectors.add(valueSelector);
+    }
+    return new ArrayOfDoublesSketchBuildAggregator(keySelector, valueSelectors, nominalEntries);
+  }
+
+  @Override
+  public BufferAggregator factorizeBuffered(final ColumnSelectorFactory metricFactory)
+  {
+    if (metricColumns == null) {
+      final BaseObjectColumnValueSelector<ArrayOfDoublesSketch> selector = metricFactory
+          .makeColumnValueSelector(fieldName);
+      if (selector instanceof NilColumnValueSelector) {
+        return new ArrayOfDoublesSketchNoOpBufferAggregator(numberOfValues);
+      }
+      return new ArrayOfDoublesSketchMergeBufferAggregator(
+        selector,
+        nominalEntries,
+        numberOfValues,
+        getMaxIntermediateSize()
+      );
+    }
+    final DimensionSelector keySelector = metricFactory
+        .makeDimensionSelector(new DefaultDimensionSpec(fieldName, fieldName));
+    if (keySelector == null || DimensionSelectorUtils.isNilSelector(keySelector)) {
+      return new ArrayOfDoublesSketchNoOpBufferAggregator(numberOfValues);
+    }
+    final List<BaseDoubleColumnValueSelector> valueSelectors = new ArrayList<>();
+    for (final String column : metricColumns) {
+      final BaseDoubleColumnValueSelector valueSelector = metricFactory.makeColumnValueSelector(column);
+      valueSelectors.add(valueSelector);
+    }
+    return new ArrayOfDoublesSketchBuildBufferAggregator(
+      keySelector,
+      valueSelectors,
+      nominalEntries,
+      getMaxIntermediateSize()
+    );
+  }
+
+  @Override
+  public Object deserialize(final Object object)
+  {
+    return ArrayOfDoublesSketchOperations.deserialize(object);
+  }
+
+  @Override
+  public Comparator<ArrayOfDoublesSketch> getComparator()
+  {
+    return COMPARATOR;
+  }
+
+  @Override
+  public Object combine(final Object lhs, final Object rhs)
```

Comment:
I think such methods are implemented so that rhs is fold into lhs, check other implementations.

Reply:
There is no way to do so with this type of sketches. A union of two sketches is always a third thing.